WcaOnRails/app/models/competition.rb

```diff
@@ -240,8 +240,10 @@ def persisted_events_id
   # The only exception to this is within tests, in which case we actually don't want to run this validation.
   validate :schedule_must_match_rounds, if: :confirmed_at_changed?, on: :update
   def schedule_must_match_rounds
-    unless has_any_round_per_event? && schedule_includes_rounds?
-      errors.add(:competition_events, I18n.t('competitions.errors.schedule_must_match_rounds'))
+    if created_at > Date.new(2019, 12, 31)
```

Comment:
Please add a comment in both places explaining what's the reasoning behind this check (and the date). Even better you can move this check `created_at > Date.new(2019, 12, 31)` into its own method with a clear name and a comment.

Reply:
Done. I also used the new function to remove the other places where "Date.new" was used.